pkg/vsphere/tasks/waiter.go

```diff
@@ -82,3 +89,54 @@ func WaitForResult(ctx context.Context, f func(context.Context) (ResultWaiter, e
 	}
 	return info, nil
 }
+
+func Retry(ctx context.Context, f func(context.Context) (ResultWaiter, error)) (*types.TaskInfo, error) {
+	rand.NewSource(time.Now().UnixNano()) //creates a more unique random
+	var err error
+	var taskInfo *types.TaskInfo
+	backoffFactor := int64(1)
+
+	//setting up timer
+	timer := time.NewTimer(time.Duration(100) * time.Millisecond)
+	if !timer.Stop() {
+		<-timer.C
+	}
+
+	for {
+		taskInfo, err = WaitForResult(ctx, f)
+
+		if err != nil {
+			return nil, err
+		}
+
+		if taskInfo.Error == nil {
+			break
+		}
+
+		if _, ok := taskInfo.Error.Fault.(types.TaskInProgressFault); !ok {
+			break
+		}
+		sleepValue := backoffFactor * (rand.Int63n(100) + int64(50))
+		timer.Reset(time.Duration(sleepValue) * time.Millisecond)
+		select {
+		case <-timer.C:
```

Comment:
turned out much simpler thanks. I have already learned a lot about the time package now. The only thing I have not used to implement this is a `ticket`